lib/composables/Form.ts

```diff
@@ -10,18 +11,33 @@ export interface Form<T extends Record<string, any>> {
   update: UpdateFunction<T>
 }
 
-export type Data<T extends Record<string, any>> = UnwrapRef<T>
-
-export type UpdateFunction<
-  T extends Record<string, any>
-> = (model: keyof Data<T>, value: any) => void
-
-export interface UseFormOptions<T extends Record<string, any>> {
+export interface UseFormOptions<T> {
   data: T | (() => T)
   rules?: Rules
 }
 
-export function useForm<T extends Record<string, any>>(options: UseFormOptions<T>): Form<T> {
+export type UpdateFunction<T extends Record<string, any> = Record<string, any>> = (
+  model: keyof Data<T>,
+  value: any
+) => void
+
+export type FormValidation<T, U extends Form<T> = Form<T>> = WithRequired<U, 'validation'>
```

Comment:
Maybe this should be `FormWithValidation`? `FormValidation` felt a bit more like it's only validation, without `data`. 👀

Reply:
I could of sworn it was 😆 ... I'll revert!